doc/designs/passkeys.md

```diff
@@ -0,0 +1,246 @@
+# Passkey authentication
+
+## Overview
+
+Traditional authentication with a password is not considered secure enough
+by many companies or government agencies. Alternate and more secure
+solutions exist, among which the use of passkeys, where the private
+key is stored on the device and the server only needs to know the public
+key.
+
+The goal of this feature is to use a passkey to authenticate a user
+against IPA.
+
+The project will be jointly developed by SSSD and IPA:
+
+- IPA provides the interface to store the user's public credentials
+- IPA provides the interface to configure passkey settings
+- SSSD performs the actual authentication
+
+## Use Cases
+
+- The administrator or the user registers a passkey into IPA, associated
+to a user account. The registration requires the insertion of the key, or
+the string output provided by SSSD registration mechanism.
+- The user is then able to authenticate to any IPA enrolled host using the
+passkey. Authentication methods include console login, desktop login,
+or su access.
```

Comment:
Please change the sentence 'Authentication methods ...' with something like "The first round of passkey integration is targeting a login to services implementing login with the help of PAM library. This includes remote access over SSH with keyboard-interactive authentication method, direct console or graphical desktop login, and authentication to PAM-protected shell services like 'su' or 'sudo'."

Reply:
I may be misunderstanding you but we won't be able to use any remote access mechanism. The authentication will happen locally with remote users.

Reply:
Ah, sorry, I jumped one step forward (krb5 auth -> use of krb5 auth).
So this should be '.. login with the help of PAM library locally on the host. This includes direct console or graphical desktop login and authentication to PAM-protected shell services ... . To access remote services a Kerberos ticket can be obtained and used against those services later.'

Reply:
That makes more sense.